crates/tinymist-query/src/upstream/complete.rs

```diff
@@ -1107,8 +1109,17 @@ impl<'a, 'w> CompletionContext<'a, 'w> {
             label,
             label_desc,
             detail,
+            bib_title,
         } in labels.into_iter().skip(skip).take(take)
         {
+            let filter_text = if let Some(bib_title) = bib_title {
+                // now format it like this, this accidentally fixes the issue
+                // with the `Match On Word Start Only` option in vscode.
+                eco_format!("{}|{}", label.as_str(), bib_title)
```

Comment:
not sure whether `filter_text` really fixes the problem, since `filter_text` is set to `label` by default.